packages/ui-components/src/components/Matrix/MatrixContext.ts

```diff
@@ -0,0 +1,89 @@
+import { Dispatch, createContext } from 'react';
+import { MatrixColumnType, MatrixRowType } from '../../types';
+import { PresentationOptions } from '@tupaia/types';
+
+type RowTitle = MatrixRowType['title'];
+
+const defaultContextValue = {
```

Comment:
This is really nice. I think that useReducer and context are a great choice here and it organises all these fiddly interactions really nicely.